src/transport/abstract-transport.js

```diff
@@ -20,6 +20,7 @@ function AbstractTransport(address, timeout, messageHandler) {
   this.awaitingPump = false;
   this.connected = false;
   this.backoffMilliseconds = 1;
+  this.nextReconnectListenerToken = 1;
```

Comment:
> When would it be necessary to track multiple newHeads subscriptions?

Answering this question in isolation (ignoring context of the rest of the comment):
`BlockStream` is built generically without making assumptions on how it will will be used.  In theory, a single `BlockStream` would be instantiated at some point but multiple callers can subscribe to notifications of new blocks.  This is useful if different parts of code want notifications independently.  One or more of those may be some kind of short-lived thing (e.g., a UI dialog that may be closed by the user) and as such they would want to unsubscribe when they are done so their code stops running and can get garbage collected.  We don't want to destroy the entire `BlockStream` object every time someone wants to unsubscribe from new block notification, so we allow as many callers to subscribe as want, and we give them tokens back so they can later unsubscribe (if they want).

If you are referring to the actual block subscription to the remote Ethereum Node (i.e., `eth_subscribe`) then there will be one per instance of `BlockStream`.  In most cases, the app will probably only have one `BlockStream` for its life and it can be treated as a singleton.  However, the code is written so this isn't strictly necessary (makes testing easier and could be useful one day).